src/Microsoft.OData.Core/ODataReaderCore.cs

```diff
@@ -396,7 +396,7 @@ public override sealed bool Read()
         public override sealed Task<bool> ReadAsync()
         {
             this.VerifyCanRead(false);
-            return this.ReadAsynchronously().FollowOnFaultWith(t => this.EnterScope(new Scope(ODataReaderState.Exception, null, null)));
+            return this.InterceptExceptionAsync(thisParam => thisParam.ReadAsynchronously());

```

Comment:
Is this and the old code functionally equivalent? How come you don't `EnterScope` here?

Reply:
I've seen i in the `InterceptException` definition below

Reply:
@habbes That's correct. The `InterceptExceptionAsync` method will take care of that